lib/SPIRV/SPIRVWriter.cpp

```diff
@@ -998,6 +998,15 @@ void LLVMToSPIRVBase::transFPGAFunctionMetadata(SPIRVFunction *BF,
         BF->addDecorate(new SPIRVDecorateStallEnableINTEL(BF));
     }
   }
+  if (MDNode *StallFree = F->getMetadata(kSPIR2MD::StallFree)) {
+    if (BM->isAllowedToUseExtension(
+            ExtensionID::SPV_INTEL_fpga_cluster_attributes)) {
+      if (getMDOperandAsInt(StallFree, 0)) {
+        BM->addCapability(spv::CapabilityFPGAClusterAttributesV2INTEL);
```

Comment:
Not sure if need to add capability explicitly. Will the test pass if you remove this line?

Reply:
Right. It's not needed.